plugin/completion.py

```diff
@@ -67,10 +67,9 @@ def handle_resolve_response(self, item: Optional[dict]) -> None:
         detail = self.format_documentation(item.get('detail') or "")
         documentation = self.format_documentation(item.get("documentation") or "")
         minihtml_content = self.get_content(documentation, detail)
-        if self.view.is_popup_visible():
-            self.update_popup(minihtml_content)
-        else:
-            self.show_popup(minihtml_content)
+        show = self.update_popup if self.view.is_popup_visible() else self.show_popup
+        # NOTE: Update/show popups from the main thread, or else the popup might make the AC widget disappear.
+        sublime.set_timeout(lambda: show(minihtml_content))
```

Comment:
Hmm, this doesn't fix the issue for `pyls`'s json module documentation, but it does fix it solargraph (I'm guessing any server that requires a `completionItem/resolve`.